### ISSUE TITLE
Add support for aliases

### ISSUE THREAD

**Issue by willnationsdev** [kortina/vscode-markdown-notes#147]
**Is your feature request related to a problem? Please describe.**
I may have very often occurring references to the same document which I want to be able to use non-piped wiki links to reference.

For example, given a `dog.md`...

    [[dogs|dog]]
    [[canine|dog]]
    [[canines|dog]]
    [[man's best friend|dog]]

It also applies for proper names that have multiple titles. For example, if discussing the television series "The Saga of Tanya the Evil"...

    [[Tanya]]
    [[Degurechaff]]
    [[Tanya Degurechaff]]
    [[Devil of the Rhine]]
    [[Protagonist]]

**Describe the solution you'd like**

I would like to bind multiple symbols to refer to the same document. It could rely on a separate prefix like the `#` and `@`-based prefixes for tags and bibtex references. For example, using ^:

    tanya_degurechaff.md
    
    # Tanya Degurechaff

    ^Tanya ^Degurechaff ^Devil-of-the-Rhine ^Protagonist ^Major-Degurechaff
    
    ---

    empire.md

    # The Empire

    The Empire is a nation with a rapid-response battalion lead by the [[Devil of the Rhine]].

Note the lack of any need to put the piped link as `[[Devil of the Rhine|Tanya Degurechaff]]`.

**Describe alternatives you've considered**

The only available alternative is to manually pass a link any time the text you wish to use remotely deviates from the actual name of the file. This is annoying if specific alternate names are used especially often.

**Comment by kortina** [comment]
Can you explain a bit more why you don't want to use `#tags` to accomplish this?

Is it because tags:documents are many:many and wiki-links:documents are currently one:one and you want wiki-links to be many:one?

**Comment by willnationsdev** [comment]
@kortina 

> Is it because tags:documents are many:many and wiki-links:documents are currently one:one and you want wiki-links to be many:one?

Yes, that is exactly it.

I have a vim shortcut to open the definition of whatever element my caret is hovering over. If I create a tag, I cannot open any such definition because there is no singular "definition" of the tag, only references. I could have a separate command to open the Peek References window, but then I don't have any quick keystroke method of immediately jumping to the file with the tag.

And, as mentioned, additional wiki links work, but require dummy files to be created that just link back to the main one.

**Comment by kortina** [comment]
Hmm, I like the idea of using `^alias` in the target document. Would that conflict with anything else in the markdown spec??

**Comment by willnationsdev** [comment]
I reviewed the [GitHub Flavored Markdown Spec](https://github.github.com/gfm/) and I don't see any use of `^`, even in its extraneous features, so I think it is safe to use.

**Comment by kortina** [comment]
One thing to note is that there would be no guarantee of uniqueness. `note1.md` and `note2.md` may both contain the alias `dog`, and you'd have to resolve to one.

For implementation, you'd probably want to do something similar at boot to how tags are handled:

https://github.com/kortina/vscode-markdown-notes/blob/master/src/NoteParser.ts#L155

Build up a mapping of `^alias` to URIs and save this in memory so you don't have to parse all of the files when you want the definition for the word under the cursor.

The other tricky thing is this may break the rendered markdown, ie, right now I think many people rely on software that will make the wiki-link clickable to the file in the preview window. I do not use this feature, but I think you'd have to do some work to get that working correctly (and actually render out the correct target filename as the link href in the preview).

**Comment by willnationsdev** [comment]
> One thing to note is that there would be no guarantee of uniqueness. `note1.md` and `note2.md` may both contain the alias `dog`, and you'd have to resolve to one.

True. I think for something like this, it would just need to be a point of documentation. That is, depending on how it is implemented/stored/refreshed, it'd either resolve to the first-one-defined, last-one-defined, or first/last of tree-order for file/directory parsing (and whichever methodology ends up being the implementation is just noted in the extension's docs). Adding config options if people want to support multiple options (though that sounds like overkill for an edge case that goes beyond intended use).

> The other tricky thing is this may break the rendered markdown

Hmm, didn't think about that aspect. But, because it isn't already part of the spec, I think if people were to combine some other Markdown parser, the parser would simply interpret the line as a standard paragraph by default.

Yes, the preview that is rendered in VS Code would need to have some sort of override to generate an appropriate link, as you describe. I'm not familiar at all with how VS Code does any of this stuff though.
